### PR TITLE
Move plugins to root

### PR DESCRIPTION
**Pull request by gondar00**
Move plugins to root
=================

# Changes 

- Added [lerna](https://github.com/lerna/lerna) to manage monorepo for plugins directory 
- Added one extra build step to build plugins in <root>/package.json
- Removed manual entry into the frontend/**/Editor/**/ files
- Moved schema files from frontend to indv. folder for each plugin
- Removed plugins from server folder

## `lerna commands`

- npx lerna create mssql -y    // creates a new package (a.k.a plugin) under plugins/packages/{name}
- npx lerna add common // adds the commonly used types and interface as a utility package (internal)
- npx lerna add knex // this adds a npm package from the npm registry to "all" packages
- npx lerna add knex --scope=mssql // this adds a npm package from the npm registry to a specific package
- npx lerna run build --stream // this builds all the packages

These commands will move into the PR: #1611 where we will be able to customize the file generation based on the application requirements (ex: creating a file inside `/docs`)

Currently, we cannot customize the file generation from [lerna](https://github.com/lerna/lerna/issues/1779) create command

### Databases: 
- [ ] Dynamodb
- [ ] Elasticsearch
- [ ] Redis
- [ ] Firestore
- [ ] Mongodb
- [ ] Postgresql
- [ ] Mysql
- [ ] Mssql

### API: 
- [ ] Airtable
- [ ] Graphql
- [ ] Stripe
- [ ] Googlesheet
- [ ] Slack
- [ ] Restapi
- [ ] Twilio
- [ ] Sendgrid
- [ ] Typesense

### Cloud: 
- [ ] S3
- [ ] Gcs

### CODE REVIEW

File: plugins/README.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# STEPS
+
+Below steps need to be run from the root repository
+
+- npx lerna create <package-name> -y
+- npx lerna add <npm-page-name> // make sure we have the same version of the npm-package-name in each package by calling lerna add from the root of the repository. ex: npx lerna add aws-sdk
+- npx lerna add <package-name> // this links <package-name> to all packages ex: npx lerna add common
```
Comment (akshaysasidrn) [inline plugins/README.md:7]: I think the above step can be mentioned after this, as trying the same command will show the version.

**Comment by akshaysasidrn** [comment]
Shouldn't we also have to update the docker files to copy and build the plugins folder too when the client and server are built separately given that plugins now is a hard dependency?

**Comment by gondar00** [comment]
> Shouldn't we also have to update the docker files to copy and build the plugins folder too when the client and server are built separately given that plugins now is a hard dependency?

Nope, Actually before client and server built we are building plugins folder, and plugins folder is added in the `package.json` of both `client` and `server` under `@tooljet/plugins` so when we run `npm install` a symlink is created and plugins folder resides inside the node_modules of the both folders,

File: plugins/README.md (hunk 1 above)
Comment (gondar00) [reply plugins/README.md:7]: ok cool

File: docker/server.Dockerfile.dev [hunk 2]
```diff
@@ -12,6 +12,7 @@ RUN mkdir -p /app
 WORKDIR /app
 
 COPY ./server/package.json ./server/package-lock.json ./
+RUN npm install -g npm@7.2.0
```
[the remark this answers is not in the document]
Comment (gondar00) [reply docker/server.Dockerfile.dev:15]: ok sure will add it as part of separate PR, 

I think there is some issue with `node:14.17.3-buster`, when I tried to re-build the images the server image was built with node `14.17.0` and in the `engines` key in package json we need `14.17.3`. 

Image in side client.dev file `node:14.17.3-alpine` installs the correct version though

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -150,6 +155,11 @@ jobs:
             ${{ runner.os }}-
       - run: apt update && apt install -y postgresql
       - run: npm i -g npm@7.20.0
+      - run: |
+          npm --prefix plugins ci
```
[the remark this answers is not in the document]
Comment (gondar00) [reply .github/workflows/ci.yml:159]: Yep this new build step introduces the above flow before we build `frontend/server` folders

3. Yeah I think `index.js` is confusing, should I rename it to `server.ts` ?

[the remark this answers is not in the document]
Comment (akshaysasidrn) [reply .github/workflows/ci.yml]: Yeah, would be better.

**Comment by gondar00** [comment]
Closing as reopened same branch in #1728

**Pull request closed without merging**